### PR TITLE
Mocks vignette

### PR DESCRIPTION
**Pull request by karima662**
Draft pull request - will add more to rmarkdown files soon

### CODE REVIEW

File: vignettes/building_mocks.Rmd [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+data, and then feeding that into one of the available print functions. In this 
+article we will only explore the use of the `print_mock_gt()` function. For 
+information on `print_to_gt()` please refer to its documentation.
+
+All tfrmt specifications are created using the `tfrmt()` function. The 
+foundation of this function are the `label`, `column`, `param` and `values` 
+parameters. From here you can specify a `body_plan` where custom 
+`frmt_structures` can be added and built upon.
+
+**Note:** layering has been used to avoid repeating code; for
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:38]: If possible, I think we should avoid the use of building template functions in this vignette. That is a more advanced topic while mocks are not.

File: vignettes/building_mocks.Rmd [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+In the following example only one format structure has been specified, where 
+the desired outcome is:
+<ul>
+<li> the count value followed by the percent value </li>
+<li> the count value is formatted to be maximum three digits </li>
+<li> the percent value is formatted to be two digits rounded to one decimal place </li>
+</ul>
+
+```{r}
+mock_tbl <- function(tfrmt_obj){
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:53]: see comment above - Is this function necessary?

File: vignettes/building_mocks.Rmd [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+```
+
+Mock tables are useful for visualising how future generated data may be 
+presented in its final form. Using the `{tfrmt}` package you can create a mock 
+table using placeholders (e.g. `xx.x`, `XXX`) or using mock data with numerical 
+values. Note that using the lower or upper case is up to your preference.
+
+Tables are generally built by creating a tfrmt specification tailored to your 
+data, and then feeding that into one of the available print functions. In this 
+article we will only explore the use of the `print_mock_gt()` function. For
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:30]:
```suggestion
article we will only explore the use of the `print_mock_gt()` function, which is specialized for printing mock tables to gt. For 
```

File: vignettes/building_mocks.Rmd [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+**Note:** layering has been used to avoid repeating code; for 
+more information please see the "Reuse and layer tfrmts" article.
+
+
+## Making a simple mock table
+
+In the following example only one format structure has been specified, where 
+the desired outcome is:
+<ul>
+<li> the count value followed by the percent value </li>
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:47]:
```suggestion
<li> the count value and percent value are combined into the same cell </li>
<li> the count value followed by the percent value </li>
```

File: vignettes/building_mocks.Rmd [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+                       "{count} {percent}",
+                       count = frmt("xxx"),
+                       percent = frmt_when("==100"~ frmt(""),
+                                           "==0"~ "",
+                                           "TRUE" ~ frmt("(xx.x%)"))))
+      ))
+}
+```
+
+The `print_mock_gt()` function has parameters `.default` and `n_cols` that
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:75]: First, we will explore how to create a mock table using this tfrmt object alone. The `print_mock_gt()` function has parameters `.default` and `n_cols`; these options inform the number of rows and columns of the mock table. In this example, we will use their default values (3) to print three row labels and three columns.

File: vignettes/building_mocks.Rmd [hunk 6]
```diff
@@ -0,0 +1,533 @@
...
+select `everything()` and remove columns that `starts_with("ord")`.
+
+You can also use `span_structure()` to add spanning labels as below.
+
+Note: You might see a warning along the lines of 
+`restarting interrupted promise evaluation`.
+
+```{r warning=FALSE}
+
+mock_tbl_adjust_cols <- function(){
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:124]: again, i'd probably avoid the use of the function and just pipe everything together

File: vignettes/building_mocks.Rmd [hunk 7]
```diff
@@ -0,0 +1,533 @@
...
+        ))
+  ) 
+}
+  
+mock_tbl_adjust_cols() %>%
+  print_mock_gt(df)
+```
+
+Alternatively you can add a span column to your dataset and allocate names for 
+each column. Subsequently, ensure you concatenate your `span` and `column`
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:148]:
```suggestion
each column. Subsequently, ensure you include both your `span` and `column` 
```

File: vignettes/building_mocks.Rmd [hunk 8]
```diff
@@ -0,0 +1,533 @@
...
+                                       "==0"~ "",
+                                       "TRUE" ~ frmt("xx.x%"))))
+    ),
+  col_plan = col_plan(
+    -starts_with("ord"))
+  ) %>%
+  print_mock_gt(df)
+```
+
+Note that if you'd like to print a select few columns from your dataset you
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:260]: Should we just show how this can be done in the col_plan instead?

File: dev/complex_mock_tables.R [hunk 9]
```diff
@@ -0,0 +1,401 @@
+#--------------------------- NO GROUPS (SAME FRMT) -----------------------------
+df <- crossing(
+  label = c("label 1", "label 2"),
+  column = c("PL", "T1", "T2"),
+  param = c("count", "percent")
+  ) %>% dplyr::arrange_all()
```
Comment (bzkrouse) [inline dev/complex_mock_tables.R:6]:
```suggestion
  ) %>% arrange(across(everything()))
```

**Review by bzkrouse — CHANGES_REQUESTED** [review]
Minor feedback/changes. This is nice and the writing is really clear! Lots of good examples in here.

File: vignettes/building_mocks.Rmd [hunk 10]
```diff
@@ -0,0 +1,533 @@
...
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r message=FALSE}
+library(tfrmt)
+library(dplyr) # for arrange_all()
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:19]:
```suggestion
library(dplyr) 
```

File: vignettes/building_mocks.Rmd [hunk 11]
```diff
@@ -0,0 +1,533 @@
...
+library(tidyr) # for crossing()
```
Comment (bzkrouse) [inline vignettes/building_mocks.Rmd:20]:
```suggestion
library(tidyr) 
```

File: vignettes/building_mocks.Rmd [hunk 12]
```diff
@@ -0,0 +1,533 @@
...
+```
+
+Mock tables are useful for visualising how future generated data may be 
+presented in its final form. Using the `{tfrmt}` package you can create a mock 
+table using placeholders (e.g. `xx.x`, `XXX`) or using mock data with numerical 
+values. Note that using the lower or upper case is up to your preference.
```
[the remark this answers is not in the document]
Comment (karima662) [reply vignettes/building_mocks.Rmd]: included in latest commit

File: vignettes/building_mocks.Rmd (hunk 2 above)
Comment (karima662) [reply vignettes/building_mocks.Rmd]: Layering removed in recent update

File: vignettes/building_mocks.Rmd (hunk 5 above)
Comment (karima662) [reply vignettes/building_mocks.Rmd]: Implemented change in latest commit

File: vignettes/building_mocks.Rmd (hunk 6 above)
Comment (karima662) [reply vignettes/building_mocks.Rmd]: see above

File: vignettes/building_mocks.Rmd (hunk 7 above)
Comment (karima662) [reply vignettes/building_mocks.Rmd]: Implemented change in latest commit

File: vignettes/building_mocks.Rmd (hunk 8 above)
Comment (karima662) [reply vignettes/building_mocks.Rmd]: Changed example to demonstrate selecting specific columns with `col_plan` but also kept note mentioning that `filter()` can be used on `df` within `print_mock_gt()` as an alternative

File: vignettes/building_mocks.Rmd [hunk 13]
```diff
@@ -0,0 +1,533 @@
...
+                                       "==0"~ "",
+                                       "TRUE" ~ frmt("xx.x%"))))
+    ),
+  col_plan = col_plan(
+    -starts_with("ord"))
+  ) %>%
+  print_mock_gt(df %>% filter(!stringr::str_detect(column, "^Risk Diff")))
+```
+
+### Same value formatting but 1 grouping level
```
[the remark this answers is not in the document]
Comment (karima662) [reply vignettes/building_mocks.Rmd:330]: see above

**Pull request merged**